### PR TITLE
Demo tests were added for the NUnit project. The project hierarchy has been updated.

### PR DESCRIPTION
**Pull request by mpushnikA1QA**
* Added demo tests for the NUnit

### CODE REVIEW

File: Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Steps/FooterFormSteps.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        [LogStep(StepType.Step)]
+        public void SaveDump()
+        {
+            footerForm.Dump.Save();
+        }
+
+        [LogStep(StepType.Assertion)]
+        public void CheckThatTheVisualElementsAreCorrect()
+        {
+            //TODO:Delete this code after correcting all errors.
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Steps/FooterFormSteps.cs]: !

File: Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Steps/TopBarMenuSteps.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        [LogStep(StepType.Step)]
+        public void ClickContactUs()
+        {
+            topBarMenu.ClickContactUs();
+        }
+
+        [LogStep(StepType.Assertion)]
+        public void ContactUsButtonIsPresent()
+        {
+            Assert.IsTrue(topBarMenu.IsContactUsButtonExist);
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Steps/TopBarMenuSteps.cs]: Please add assertion message

File: Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Steps/TopBarMenuSteps.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+using NUnit.Framework;
+using Aquality.Selenium.Template.CustomAttributes;
+
+namespace Aquality.Selenium.Template.NUnit.Steps
+{
+    public class TopBarMenuSteps
+    {
+        private readonly TopBarMenu topBarMenu = new TopBarMenu();
+        private const string ServicesTabItem = "Services";
+        private readonly string[] headerTabItems = { "Services", "Approach", "Portfolio", "Blog", "Company" };
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Steps/TopBarMenuSteps.cs]: should be static I believe, and named starting with an uppercase letter. both headerTabItems  and servicesTitleElements

File: Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Tests/CustomImageComparatorTest.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+using NUnit.Allure.Core;
+using NUnit.Framework;
+
+namespace Aquality.Selenium.Template.NUnit.Tests
+{
+    [AllureNUnit]
+    [AllureSuite("CustomImageComparatorTest")]
+    public class CustomImageComparatorTest : BaseTest
+    {
+        const float CustomThresholdValue = 0.1f;
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Tests/CustomImageComparatorTest.cs]: please add private access modifier

File: Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/runtimeconfig.template.json [hunk 5]
```diff
@@ -0,0 +1,5 @@
+{
+  "configProperties": {
+    "System.Drawing.EnableUnixSupport": true
+  }
+}
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/runtimeconfig.template.json]: please add an empty line at the end of file

File: Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Resources/JavaScripts/GetFullPageHeight.js [hunk 6]
```diff
@@ -0,0 +1 @@
+﻿return document.body.scrollHeight
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template.NUnit/Resources/JavaScripts/GetFullPageHeight.js]: please add an empty line at the end of file

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace Aquality.Selenium.Template.Extensions
+{
+    public static class ElementFactoryExtensions
+    {
+        public static IList<T> GetNotEmptyElementList<T>(this IElementFactory elementFactory, By by, string itemName) where T : IElement
+        {
+            try
+            {
+                return elementFactory.FindElements<T>(by, expectedCount: ElementsCount.MoreThenZero);
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs:16]: Do we still need this method, or was it some workaround for Aquality.Selenium issue?

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/EnumExtension.cs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+namespace Aquality.Selenium.Template.Extensions
+{
+    public static class EnumExtension
+    {
+        public static string GetEnumDescription(this Enum enumValue)
+        {
+            string value = enumValue.ToString();
+            FieldInfo field = enumValue.GetType().GetField(value);
+            object[] objs = field.GetCustomAttributes(typeof(DescriptionAttribute), false);     //Getting the description attribute
+            if (objs == null || objs.Length == 0)                                               //If the attribute is not found, it returns the default value
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/EnumExtension.cs]: please add curly braces for if statement, and use var for variables declaration

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/FooterForm.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+        private ILabel ContactsLabel => FormElement.FindChildElement<ILabel>(By.XPath("//div[contains(@class,'footer__contacts')]"), "Contacts");
+
+        private ILabel SubscribeLabel => FormElement.FindChildElement<ILabel>(By.XPath("//div[contains(@class,'footer__subscribe')]"), "Subscribe");
+
+        public FooterForm() : base(By.TagName("footer"), "Footer form")
+        {
+        }
+
+        public bool IsLogoPresent => LogoLabel.State.WaitForDisplayed(AqualityServices.Get<ICustomTimeoutConfiguration>().ElementAppear);
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Forms/FooterForm.cs]: As we use this reference `AqualityServices.Get<ICustomTimeoutConfiguration>().ElementAppear` 3 times a row, it's better to move it into the `private static readonly TimeSpan ElementPresenceTimeout`

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Utilities/BrowserUtils.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
+﻿using OpenQA.Selenium;
+using static Aquality.Selenium.Browsers.AqualityServices;
+
+namespace Aquality.Selenium.Template.Utilities
+{
+    public static class BrowserUtils
+    {
+        public static void OpenInNewTabAndSwitch(string url)
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Utilities/BrowserUtils.cs]: Maybe we can remove this method and just pass the parameter `switchToNew: true`?  It's already exist in Aquality.Selenium:
image

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs (hunk 7 above)
Comment (mpushnikA1QA) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs:16]: This method is like a demonstration of the logic of creating an extension. This method is used.
If an element is not found by the condition, then there will be an error TimeoutException without correct message.

Comment (mpushnikA1QA) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs:16]: image

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Aquality.Selenium.Template.csproj [hunk 11]
```diff
@@ -16,14 +16,14 @@
...
+    <PackageReference Include="Aquality.Selenium" Version="4.9.1" />
+    <PackageReference Include="AspectInjector" Version="2.8.1" />
     <PackageReference Include="Humanizer" Version="2.14.1" />
     <PackageReference Include="NUnit.Allure" Version="1.2.1.1" />
     <PackageReference Include="AqualityTracking.Integrations.Core" Version="1.0.1" />
     <PackageReference Include="NUnit" Version="3.13.3" />
-    <PackageReference Include="RestSharp" Version="108.0.1" />
-    <PackageReference Include="System.Drawing.Common" Version="6.0.0" />
+    <PackageReference Include="RestSharp" Version="108.0.3" />
+    <PackageReference Include="System.Drawing.Common" Version="7.0.0" />
```
Comment (mialeska) [inline Aquality.Selenium.Template/Aquality.Selenium.Template/Aquality.Selenium.Template.csproj]: please reverd to 6.0.0

**Review by mialeska — CHANGES_REQUESTED** [review]

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Aquality.Selenium.Template.csproj (hunk 11 above)
Comment (mpushnikA1QA) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Aquality.Selenium.Template.csproj]: The version was returned in a subsequent commit - https://github.com/aquality-automation/aquality-selenium-dotnet-template/pull/14/commits/ebf288194f0d90fd32dd1f2343d48873e6ec34c4

**Review by mialeska — APPROVED** [review]

File: Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs (hunk 7 above)
Comment (mialeska) [reply Aquality.Selenium.Template/Aquality.Selenium.Template/Extensions/ElementFactoryExtensions.cs:16]: If it's a bug please create issue in Aquality.Selenium.Core, we will return to this later

**Pull request merged**